### PR TITLE
Add Github Actions to publish and tag btcpay ln libraries

### PR DESCRIPTION
**Pull request by Kukks**
This will check all our projects which are nuget libs, extract the version, check if it is on nuget, and if not pack and publish and tag the commit with the version. 

Additionall,y will also publish in parallel to Github package registry so that there is an alternative source for the libs.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                    INCLUDE_SYMBOLS: true
+                    VERSION_REGEX: ^\s*<Version>(.*)<\/Version>\s*$
+                    TAG_COMMIT: true
+                    TAG_FORMAT: Eclair/v*
+            #BTCPayServer.Lightning.LNbank nuget    
+            -   name: Publish NuGet for BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank
+                if: ${{ github.ref == 'refs/heads/master' }} # Publish only when the push is on master
+                uses: Rebel028/publish-nuget@v2.7.0
+                with:
+                    PROJECT_FILE_PATH: src/BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank/BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank.csproj
```
Comment (dennisreimann) [inline .github/workflows/ci.yml:119]:
```suggestion
                    PROJECT_FILE_PATH: src/BTCPayServer.Lightning.LNbank/BTCPayServer.Lightning.LNbank.csproj
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                    PACKAGE_NAME: BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank
+                    INCLUDE_SYMBOLS: true
+                    VERSION_REGEX: ^\s*<Version>(.*)<\/Version>\s*$
+                    TAG_COMMIT: true
+                    TAG_FORMAT: BTCPayServer.Lightning.LNbank/v*
+            -   name: Publish Github Package Registry for BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank
+                if: ${{ github.ref == 'refs/heads/master' }} # Publish only when the push is on master
+                uses: Rebel028/publish-nuget@v2.7.0
+                with:
+                    PROJECT_FILE_PATH: src/BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank/BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank.csproj
```
Comment (dennisreimann) [inline .github/workflows/ci.yml:130]:
```suggestion
                    PROJECT_FILE_PATH: src/BTCPayServer.Lightning.LNbank/BTCPayServer.Lightning.LNbank.csproj
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                    NUGET_SOURCE: "https://nuget.pkg.github.com/btcpayserver"
+                    NUGET_KEY: ${{secrets.GH_TOKEN}}
+                    PACKAGE_NAME: BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank
```
Comment (dennisreimann) [inline .github/workflows/ci.yml:133]:
```suggestion
                    PACKAGE_NAME: BTCPayServer.Lightning.LNbank
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                    TAG_COMMIT: true
+                    TAG_FORMAT: Eclair/v*
+            #BTCPayServer.Lightning.LNbank nuget    
+            -   name: Publish NuGet for BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank
+                if: ${{ github.ref == 'refs/heads/master' }} # Publish only when the push is on master
+                uses: Rebel028/publish-nuget@v2.7.0
+                with:
+                    PROJECT_FILE_PATH: src/BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank/BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank.csproj
+                    NUGET_KEY: ${{secrets.NUGET_KEY}}
+                    PACKAGE_NAME: BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank
```
Comment (dennisreimann) [inline .github/workflows/ci.yml:121]:
```suggestion
                    PACKAGE_NAME: BTCPayServer.Lightning.LNbank
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+                    PROJECT_FILE_PATH: src/BTCPayServer.Lightning.Eclair/BTCPayServer.Lightning.Eclair.csproj
+                    NUGET_SOURCE: "https://nuget.pkg.github.com/btcpayserver"
+                    NUGET_KEY: ${{secrets.GH_TOKEN}}
+                    PACKAGE_NAME: BTCPayServer.Lightning.Eclair
+                    INCLUDE_SYMBOLS: true
+                    VERSION_REGEX: ^\s*<Version>(.*)<\/Version>\s*$
+                    TAG_COMMIT: true
+                    TAG_FORMAT: Eclair/v*
+            #BTCPayServer.Lightning.LNbank nuget    
+            -   name: Publish NuGet for BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank
```
Comment (dennisreimann) [inline .github/workflows/ci.yml:115]:
```suggestion
            -   name: Publish NuGet for BTCPayServer.Lightning.LNbank
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+                uses: Rebel028/publish-nuget@v2.7.0
+                with:
+                    PROJECT_FILE_PATH: src/BTCPayServer.Lightning.LNbank/BTCPayServer.Lightning.LNbank.csproj
+                    NUGET_KEY: ${{secrets.NUGET_KEY}}
+                    PACKAGE_NAME: BTCPayServer.Lightning.LNbank
+                    INCLUDE_SYMBOLS: true
+                    VERSION_REGEX: ^\s*<Version>(.*)<\/Version>\s*$
+                    TAG_COMMIT: true
+                    TAG_FORMAT: BTCPayServer.Lightning.LNbank/v*
+            -   name: Publish Github Package Registry for BTCPayServer.Lightning.BTCPayServer.Lightning.LNbank
```
Comment (dennisreimann) [inline .github/workflows/ci.yml:126]:
```suggestion
            -   name: Publish Github Package Registry for BTCPayServer.Lightning.LNbank
```

**Review by dennisreimann — APPROVED** [review]

**Comment by NicolasDorier** [comment]
I removed the `PushNuget.ps1` scripts, and I removed the push to github nuget package repo.
I don't see why it is useful, and it is likely to give us headache of maintenance down the line.
If one day nuget stops working, we can find another solution then.

However, there should be 8 packages, but I can see only 6 that are pushed, so two are missing.

**Pull request merged**